libdnf5/base/log_event.cpp

```diff
@@ -153,23 +153,52 @@ std::string LogEvent::to_string(
                 } else {
                     return ret.append(utils::sformat(_("No match for group package: {}"), *spec));
                 }
+            } else if (goal_action_is_replay(action)) {
+                return ret.append(
+                    utils::sformat(_("Cannot perform {}, no match for: {}."), goal_action_to_string(action), *spec));
             }
             return ret.append(utils::sformat(_("No match for argument: {}"), *spec));
         case GoalProblem::NOT_FOUND_IN_REPOSITORIES:
             return ret.append(utils::sformat(
                 _("No match for argument '{0}' in repositories '{1}'"),
                 *spec,
                 utils::string::join(settings->get_to_repo_ids(), ", ")));
-        case GoalProblem::NOT_INSTALLED:
+        case GoalProblem::NOT_INSTALLED: {
+            if (goal_action_is_replay(action)) {
+                return ret.append(utils::sformat(
+                    _("Cannot perform {} for {} '{}' becasue it is not installed."),
+                    goal_action_to_string(action),
+                    transaction_item_type_to_string(*spec_type),
+                    *spec));
+            }
             return ret.append(utils::sformat(_("Packages for argument '{}' available, but not installed."), *spec));
+        }
         case GoalProblem::NOT_INSTALLED_FOR_ARCHITECTURE:
             return ret.append(utils::sformat(
                 _("Packages for argument '{}' available, but installed for a different architecture."), *spec));
         case GoalProblem::ONLY_SRC:
+            if (goal_action_is_replay(action)) {
+                return ret.append(utils::sformat(
+                    _("Cannot perform {} because '{}' matches only source packages."),
+                    goal_action_to_string(action),
+                    *spec));
+            }
             return ret.append(utils::sformat(_("Argument '{}' matches only source packages."), *spec));
         case GoalProblem::EXCLUDED:
+            if (goal_action_is_replay(action)) {
+                return ret.append(utils::sformat(
+                    _("Cannot perform {} because '{}' matches only excluded packages."),
```

Comment:
May I ask you to mark an order of arguments `{1}`. It will allow to change order for translators. In this case what about to provide a context message for translators?

Reply:
Added the marks but I don't know what are context messages? Can we perhaps add them separately?